### PR TITLE
Spinner icon button modifier

### PR DESCRIPTION
**Pull request by claredevine**
Spinner icon button.
When clicked cursor will change to wait.

N.b. this is currently being trialled in FAV to see the reaction from customers.

## Description
Spinner sat inside a button that is right of the text.

## Motivation and Context
Flow Selecta - FAV

## How Has This Been Tested?
Tested locally

```
<div class="o-container u-padding-top">
        <button className="c-btn c-btn--primary">
          Loading...
          <strong className='c-spinner c-btn__spinner' role='progressbar'></strong>
        </button>
</div>
```
## Screenshots (if appropriate)
screen shot 2017-03-21 at 17 21 21

## Types of Changes
What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Browser Support
- [x] Chrome
- [x] Edge
- [x] Firefox
- [x] IE9
- [x] IE10
- [x] IE11
- [x] Opera
- [x] Safari
- [x] Mobile Devices
- [x] Screen-readers (e.g. JAWS, Apple VoiceOver)

## Checklist
- [ ] All code includes full inline documentation (as per the [template](https://github.com/sky-uk/toolkit-core/blob/master/_template.scss)).
- [ ] All code conforms to the Toolkit [coding style](https://github.com/sky-uk/toolkit/wiki/Coding-Style).
- [ ] All code conforms to [WCAG 2.0 level AA Accessibility Guidelines](https://www.w3.org/TR/WCAG20/).
- [ ] New functions and mixins have appropriate tests.
- [ ] All new and existing tests passed.
- [ ] I have added instructions on how to test my changes.
- [ ] Package version updated.
- [ ] CHANGELOG.md updated.

### CODE REVIEW

File: components/_buttons.scss [hunk 1]
```diff
@@ -197,3 +197,25 @@ $btn-animation-speed-fast: $global-animation-speed-fast !default;
...
+/**
+* For when icons are used instead of typography.
+*/
+.c-btn--spinner {
+  padding-left: 40px;
+  padding-right: 40px;
+    &:before {
+      content: '';
+      display: inline-block;
+      vertical-align: -6.5px;
```
Comment (joebell93) [inline components/_buttons.scss]: What's the reasoning for using this specific value?

**Comment by joebell93** [comment]
Could you show some example HTML please 😊 ?

File: components/_buttons.scss (hunk 1 above)
Comment (claredevine) [inline components/_buttons.scss]: The `c-btn--spinner` seems to override the size of `c-btn` this ensures that both are the same size 😄

**Comment by claredevine** [comment]
@joebell93 yeah, just adding that feature now, won't be long - works in dev tools so it won't break. Will get onto the HTML snippet now

**Comment by joebell93** [comment]
@claredevine can we check this is all good from a UX perspective before we go forwards?

I was kinda under the impression it was an icon on the right kinda deal.

**Comment by claredevine** [comment]
of course 👍

**Comment by StefanMcCready** [comment]
@danieljbryson was taking a look into an icon concept with buttons as I mentioned this idea a couple of weeks back which led to the Icon idea. Worth bringing up in design web and check on how far Dan got with it.

**Comment by claredevine** [comment]
@joebell93 From Liam's comments were going to have the button only to be used with text so that the customer can see what check they have selected.

File: components/_buttons.scss (hunk 1 above)
Comment (claredevine) [inline components/_buttons.scss]: @joebell93 removed it

**Comment by StefanMcCready** [comment]
One final but of food for thought. Seen as though this would probably be used in a certain UI state. Could we not use a state modifier? Something like `.is-waiting` or is `.is-loading`

**Comment by claredevine** [comment]
This could be thought about in the future when we revisit it 😃

**Comment by joebell93** [comment]
Looks good to me @claredevine as `c-btn__spinner` 👌 

Just need to make clear in the documentation that [`.is-complete`](https://github.com/sky-uk/toolkit-ui/blob/master/components/_spinner.scss#L44) is added when finished!

We may have to make some adjustments to the padding etc for this in `.c-btn__spinner.is-complete`

**Comment by claredevine** [comment]
@joebell93 what do you mean by `.is-complete`? if the button class is set to `.is-disabled` then the button will be greyed out (which is what we want).
`.is-complete` would mean more or less replicating the code from `.is-disabled` - which we kinda dont want to do

**Comment by joebell93** [comment]
Closing due to the [Monorepo](https://github.com/sky-uk/toolkit) move

**Pull request closed without merging**